docs/new-feature-tutorial.md

```diff
@@ -88,29 +88,51 @@ documentation Zulip has, see [Documentation](README.html).
 
 This example describes the process of adding a new setting to Zulip: a
 flag that restricts inviting new users to admins only (the default
-behavior is that any user can invite other users). It is based on an
-actual Zulip feature, and you can review [the original commit in the
-Zulip git
-repo](https://github.com/zulip/zulip/commit/5b7f3466baee565b8e5099bcbd3e1ccdbdb0a408).
-(Note that Zulip has since been upgraded from Django 1.6 to 1.8, so the
-migration format has changed.)
+behavior is that any user can invite other users). This flag is an actual
+Zulip feature. You can review
+[the original commit](https://github.com/zulip/zulip/commit/5b7f3466baee565b8e5099bcbd3e1ccdbdb0a408)
+in the Zulip repo.
+Note that the code involved in adding a realm feature has been refactored
+since this feature was created, and Zulip has since been upgraded from
+Django 1.6 to 1.8.
 
 ### Update the model
 
 First, update the database and model to store the new setting. Add a new
 boolean field, `invite_by_admins_only`, to the Realm model in
 `zerver/models.py`.
 
-``` diff
---- a/zerver/models.py
-+++ b/zerver/models.py
-@@ -139,6 +139,7 @@ class Realm(ModelReprMixin, models.Model):
-     restricted_to_domain = models.BooleanField(default=True) # type: bool
-     invite_required = models.BooleanField(default=False) # type: bool
-+    invite_by_admins_only = models.BooleanField(default=False) # type: bool
-     create_stream_by_admins_only = models.BooleanField(default=False) # type: bool
-     mandatory_topics = models.BooleanField(default=False) # type: bool
-```
+    class Realm(ModelReprMixin, models.Model):
+       # ...
+       invite_by_admins_only = models.BooleanField(default=False) # type: bool
+       # ...
+
+The Realm model also contains an attribute, `property_types`, which other
+functions use to handle realm updates appropriately (more on this
+process below). The attribute is a dictionary, where the key is the
+name of the realm field and the value is the field's type. Add the
+new field to the `property_types` dictionary.
+
+    # Define the types of the various automatically managed properties
+        property_types = dict(
+            # ...
+            invite_by_admins_only=bool,
+            # ...
+
+Note: the majority of realm fields can be included in `property_types`
+and processed in bulk in the code that handles realm property updates.
+However, there are some properties that should be processed separately.
+For example, the realm `authentication_methods` attribute
+is a bitfield and needs additional code for validation and updating.
+The `allow_message_editing` and `message_content_edit_limit_seconds`
+fields depend on one another, so they are also handled separately and
+not included in `property_types`.
+
+When creating a realm property that is not a boolean, Text or
```

Comment:
I guess this is something that we'll be able to change back to `str` when we kill Python 2 support.